### PR TITLE
Add UT for pkg/controllers/status/cluster_status_controller_test.go

### PR DESCRIPTION
**Pull request by realnumber666**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Promote UT coverage

**Which issue(s) this PR fixes**:
Part of #3327 

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

**Review by zishen — COMMENTED** [review]
Good for controller test.

**Comment by XiShanYongYe-Chang** [comment]
Thanks~
/assign

File: pkg/controllers/status/cluster_status_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,1121 @@
...
+				GenericInformerManager: genericmanager.GetInstance(),
+				TypedInformerManager:   typedmanager.GetInstance(),
+				ClusterClientOption: &util.ClientOption{
+					QPS:   5,
+					Burst: 10,
+				},
+				ClusterClientSetFunc: util.NewClusterClientSet,
+			}
+
+			if tt.existingCluster {
```
Comment (XiShanYongYe-Chang) [inline pkg/controllers/status/cluster_status_controller_test.go:92]: We can use `tt.Cluster != nil` to replace the judgment, so we can omit the `existingCluster` parameter.

File: pkg/controllers/status/cluster_status_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,1121 @@
...
+				if err := c.Client.Create(context.Background(), tt.cluster); err != nil {
+					t.Fatalf("Failed to create cluster: %v", err)
+				}
+			}
+
+			// Run the reconcile function.
+			req := controllerruntime.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      "test-cluster",
+					Namespace: "default",
```
Comment (XiShanYongYe-Chang) [inline pkg/controllers/status/cluster_status_controller_test.go:103]: How about setting these two values as input variables?

File: pkg/controllers/status/cluster_status_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,1121 @@
...
+		},
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{Namespace: "ns1", Name: "secret1"},
+			Data: map[string][]byte{
+				clusterv1alpha1.SecretTokenKey: []byte("token"),
+			},
+		}).Build()
+	clusterClient, _ := util.NewClusterClientSet("test", hostClient, nil)
+	clusterClientSet.KubeClient = clusterClient.KubeClient
+	return clusterClientSet
```
Comment (XiShanYongYe-Chang) [inline pkg/controllers/status/cluster_status_controller_test.go:143]: Can here directly return `clusterClient`?

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Good job!

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**